src/components/LabEnvironment/LabEnvironment.tsx

```diff
@@ -61,6 +75,59 @@ export class LabEnvironment extends Component<LabEnvironmentProps, LabEnvironmen
     return this.props.userLab.status !== 'Started';
   }
 
+  setLabEndDateTimeInterval = (userLab: UserLab) => {
+    this.state.interval = window.setInterval(() => {
+      this.setState({ labEndDateTime: userLab.endDateTime! });
+    }, 1000);
+  }
+
+  handleExtendEndDateTime = async () => {
+    const updatedUserLab = await updateEndDateTime(this.props.userLab.id);
+    window.clearInterval(this.state.interval);
+    this.setState({ labEndDateTime: updatedUserLab.endDateTime! });
+    this.setLabEndDateTimeInterval(updatedUserLab);
+  };
+
+  showExtendButton = () => {
+    const remainingLabTime = getLuxonObjectFromString(this.state.labEndDateTime!).toLocal().diffNow().as('second');
+    // shows a pop up if the lab times up instead of providing an extend option
+    if (getRemainingLabTime(this.state.labEndDateTime) === 'Times up!') {
+      window.setTimeout(() => {
+        window.location.replace(RoutePaths.myModules);
+      }, 15000);
+      return (
+        <Layout>
```

Comment:
Why is this wrapped with layout? LabEnvironment is already wrapped in a Layout component.